netbox/extras/middleware.py

```diff
@@ -56,11 +60,16 @@ def __call__(self, request):
         # Process the request
         response = self.get_response(request)
 
-        # Record object changes
+        # Perform change logging and fire Webhook signals
         for obj, action in _thread_locals.changed_objects:
-            if obj.pk:
+            # Log object changes
+            if obj.pk and hasattr(obj, 'log_change'):
```

Comment:
I want to say there was a specific reason we need to check for `obj.pk` here but I could be wrong. Can't think of a situation where we'd get a post_save or post_delete signal for an object _without_ a primary key. (The pk stays with the object in memory even after being deleted from the database.) We might not need it.